great_expectations/experimental/datasource/fabric.py

```diff
@@ -0,0 +1,394 @@
+    fully_qualified_columns: Optional[bool] = None
+    num_rows: Optional[int] = None
+    pandas_convert_dtypes: bool = True
+    use_xmla: bool = False
+
+
+class PowerBITable(_PowerBIAsset):
+    """Microsoft PowerBI Table."""
+
+    _reader_method: ClassVar[FabricReaderMethods] = "read_table"
+
+    type: Literal["powerbi_table"] = "powerbi_table"
+    table: str
+    fully_qualified_columns: bool = False
+    num_rows: Optional[int] = None
+    multiindex_hierarchies: bool = False
+    pandas_convert_dtypes: bool = True
+
+
+# This improves our error messages by providing a more specific type for pydantic to validate against
+# It also ensure the generated jsonschema has a oneOf instead of anyOf field for assets
+# https://docs.pydantic.dev/1.10/usage/types/#discriminated-unions-aka-tagged-unions
+AssetTypes = Annotated[
+    Union[PowerBITable, PowerBIMeasure, PowerBIDax],
+    pydantic.Field(discriminator="type"),
+]
+
+
+class FabricPowerBIDatasource(Datasource):
+    """
+    Microsoft Fabric Datasource.
+
+    https://pypi.org/project/semantic-link/
+    """
+
+    # class var definitions
+    asset_types: ClassVar[List[Type[DataAsset]]] = [
+        PowerBIDax,
+        PowerBIMeasure,
+        PowerBITable,
+    ]
+    # any fabric datsource specific fields should be added to this set
+    # example a connection_string field or a data directory field
+    _EXTRA_EXCLUDED_EXEC_ENG_ARGS: ClassVar[set] = {"workspace", "dataset"}
+
+    # right side of the operator determines the type name
+    # left side enforces the names on instance creation
+    type: Literal["fabric_powerbi"] = "fabric_powerbi"
+    assets: List[AssetTypes] = []
+
+    # fabric datasource specific fields
+    workspace: Optional[Union[uuid.UUID, str]] = None
+    dataset: Union[uuid.UUID, str]
+
+    @property
+    @override
+    def execution_engine_type(self) -> Type[PandasExecutionEngine]:
+        """Return the PandasExecutionEngine unless the override is set"""
+        from great_expectations.execution_engine.pandas_execution_engine import (
+            PandasExecutionEngine,
+        )
+
+        return PandasExecutionEngine
+
+    @override
+    def test_connection(self, test_assets: bool = True) -> None:
+        """Test the connection for the FabricPowerBIDatasource.
+
+        Args:
+            test_assets: If assets have been passed to the Datasource, whether to test them as well.
+
+        Raises:
+            TestConnectionError: If the connection test fails.
+        """
+        if not self._running_on_fabric():
+            raise TestConnectionError(
+                "Must be running Microsoft Fabric to use this datasource"
+            )
+
+        try:
+            from sempy import fabric  # noqa: F401 # test if fabric is installed
+        except Exception as import_err:
+            raise TestConnectionError(
+                "Could not import `sempy.fabric`\npip install semantic-link"
```

Comment:
you can replace with semantic-link with semantic-link-sempy

Reply:
Done